pyeudiw/trust/handler/x509.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from typing import Union
+from pyeudiw.trust.handler.interface import TrustHandlerInterface
+from pyeudiw.trust.model.trust_source import TrustSourceData, TrustParameterData
+from pyeudiw.trust.handler.exceptions import InvalidTrustHandlerConfiguration
+from pyeudiw.x509.verify import (
+    verify_x509_attestation_chain, 
+    get_expiry_date_from_x5c, 
+    der_list_to_pem_list, 
+    pem_list_to_der_list, 
+    get_x509_info
+)
+
+logger = logging.getLogger(__name__)
+
+class X509Hanlder(TrustHandlerInterface):
+    """
+    X509Handler is a trust handler implementation that extracts trust material from x509 certificates.
+    """
+    def __init__(
+        self, 
+        client_id: str, 
+        relying_party_certificate_chains_by_ca: dict[str, Union[list[bytes], list[str]]],
+        private_keys: list[dict[str, str]],
+        client_id_scheme: str = "x509_san_uri",
+        **kwargs
+    ):  
+        self.client_id = client_id
+        self.client_id_scheme = client_id_scheme
+
+        if not relying_party_certificate_chains_by_ca:
+            raise InvalidTrustHandlerConfiguration("No x509 certificate chains provided")
+
+        self.relying_party_certificate_chains_by_ca = {}
+
+        for k, v in relying_party_certificate_chains_by_ca.items():
+            root_dns_name = get_x509_info(v[-1])
+            
+            if not root_dns_name in k:
+                raise InvalidTrustHandlerConfiguration(f"Invalid x509 certificate: expected {k} got {root_dns_name}")
```

Comment:
```suggestion
                raise InvalidTrustHandlerConfiguration(f"Invalid x509 certificate: expected {k} got {root_dns_name} instead of {k}")
```